env/CMakeLists.txt

```diff
@@ -36,15 +35,15 @@ ExternalProject_Add(
 # -DMLIR_ENABLE_EXPENSIVE_PATTERN_API_CHECKS=ON set if debug
 ExternalProject_Add(
     llvm-project
-    PREFIX ${TTMLIR_TOOLCHAIN_DIR}
```

Comment:
I'm not sure how this can work without setting the prefix.  This means the project will be built and linked with the default prefix which is probably `/usr/local`.

Reply:
Ok so I misunderstood `PREFIX` in the context of `ExternalProject_Add`.  Actually what I meant to write here was:
```
INSTALL_DIR ${TTMLIR_TOOLCHAIN_DIR}
```

Can we try that instead?  And then we can try:
```
cmake --install env/build
```

This feels like the proper way of doing things.

Reply:
Yes, I'll look into it. Right now it has -DCMAKE_INSTALL_PREFIX=${TTMLIR_TOOLCHAIN_DIR} and build target also installs in the end.
I'll see if we can split build and install.

Reply:
Ah good point yeah ` -DCMAKE_INSTALL_PREFIX=${TTMLIR_TOOLCHAIN_DIR}` should work.  Ok see if `cmake --install env/build` can work instead of doing the manual copy, the install should install the minimum set of things needed so it could theoretically cut down on the image size.

Reply:
Ok sorry, last point, were does this actually end up building the toolchain?  Just wondering how this will impact locally building the toolchain, i.e. not in docker.

Reply:
Build happens in env/build folder.
For example object files are here:

env/build/llvm-project-prefix/src/llvm-project-build/lib/Target/AMDGPU/CMakeFiles/LLVMAMDGPUCodeGen.dir/R600Subtarget.cpp.o
...
env/build/stablehlo-prefix/src/stablehlo-build/stablehlo/transforms/CMakeFiles/obj.StablehloPasses.dir/StablehloLegalizeQuantizedOpToQDQ.cpp.o

Archive here

env/build/llvm-project-prefix/src/llvm-project-build/lib/libLLVMLinker.a

And libs here

env/build/llvm-project-prefix/src/llvm-project-build/lib/libMLIR.so

After build is done, headers, .a  and .so are installed in /opt/ttmlir-toolchain/

/opt/ttmlir-toolchain/lib/libMLIR.so

Reply:
It looks like ExternalProject_Add automatically installs dependencies on build. There is a way to disable this by setting INSTALL_COMMAND = "", but it won't give us "cmake --install env/build" installation. 

ExternalProject_Add
The ExternalProject_Add() function creates a custom target to drive download, update/patch, configure, build, install and test steps of an external project

https://cmake.org/cmake/help/v3.10/module/ExternalProject.html

My understanding is that we should keep things that are required for building the project in ExternalProject_Add and let them build and install them before building the project.

We should add install() step that installs tt-mlir output binaries and headers, so in future downstream projects can install.

https://cmake.org/cmake/help/latest/command/install.html

Reply:
Yeah agreed, yeah let's leave the default `ExternalProject_Add` behavior i.e. let it auto-install. Ok cool this sounds good, thanks!